### PR TITLE
Tests around Atomic host Provisioning on external libvirt

### PR DESCRIPTION
**Pull request by sghai**
- Includes two tests around atomic host
a) Provision atomic host on libvirt
b) delete provisioned atomic host

Since kickstart installer for atomic is not available via CDN so defined a setting in properties file. Will update the jenkins with this change.

`Test results:`

```console
 robottelo]$ py.test -v  tests/foreman/ui/test_host.py::AtomicHostTestCase -k test_positive_delete_atomic_host
================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.5, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
plugins: cov-2.3.1, xdist-1.15.0
collected 7 items 

tests/foreman/ui/test_host.py::AtomicHostTestCase::test_positive_delete_atomic_host PASSED

============================================= 6 tests deselected by '-ktest_positive_delete_atomic_host' =============================================
====================================================== 1 passed, 6 deselected in 304.21 seconds ==================================================

robottelo]$ py.test -v tests/foreman/ui/test_host.py::AtomicHostTestCase -k test_positive_provision_atomic_host
================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.5, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /usr/bin/python
cachedir: .cache
rootdir: /home/sghai/QE/robottelo, inifile: 
plugins: cov-2.3.1, xdist-1.15.0
collected 7 items 

tests/foreman/ui/test_host.py::AtomicHostTestCase::test_positive_provision_atomic_host PASSED

=========================================== 6 tests deselected by '-ktest_positive_provision_atomic_host' ============================================
====================================================== 1 passed, 6 deselected in 295.09 seconds ======================================================
```

### CODE REVIEW

File: tests/foreman/ui/test_host.py [hunk 1]
```diff
@@ -536,7 +545,284 @@ def test_positive_update_with_user(self):
...
+        11. Associates org, location and OS with provisioning and PXE templates
+        12. Search for x86_64 architecture
+        13. Associate arch, partition table, provisioning/PXE templates with OS
+        14. Search for existing Atomic media or create new otherwise and
+            associate org/location
+        15. Create new host group with all required entities
+        """
+        super(AtomicHostTestCase, cls).setUpClass()
+        # Create a new Organization and Location
+        cls.org = entities.Organization(name=gen_string('alpha')).create()
```
Comment (oshtaier) [inline tests/foreman/ui/test_host.py]: `cls.org = entities.Organization().create()`

File: tests/foreman/ui/test_host.py [hunk 2]
```diff
@@ -536,7 +545,284 @@ def test_positive_update_with_user(self):
...
+        cls.org_name = cls.org.name
+        cls.loc = entities.Location(
```
Comment (oshtaier) [inline tests/foreman/ui/test_host.py]: `cls.loc = entities.Location(organization=[cls.org]).create()`

File: tests/foreman/ui/test_host.py [hunk 3]
```diff
@@ -536,7 +545,284 @@ def test_positive_update_with_user(self):
...
+            name=gen_string('alpha'),
+            organization=[cls.org],
+        ).create()
+        cls.loc_name = cls.loc.name
+        # Create a new Life-Cycle environment
+        cls.lc_env = entities.LifecycleEnvironment(
```
Comment (oshtaier) [inline tests/foreman/ui/test_host.py]: `cls.lc_env = entities.LifecycleEnvironment(organization=cls.org).create()`

File: tests/foreman/ui/test_host.py [hunk 4]
```diff
@@ -536,7 +545,284 @@ def test_positive_update_with_user(self):
...
+            # OrgId is passed as data in API hence str
+            org_id=str(cls.org.id),
+            product=cls.rh_ah_repo['product'],
+            repo=cls.rh_ah_repo['name'],
+            reposet=cls.rh_ah_repo['reposet'],
+            releasever=cls.rh_ah_repo['releasever'],
+        )
+        # Increased timeout value for repo sync
+        cls.old_task_timeout = entity_mixins.TASK_TIMEOUT
+        entity_mixins.TASK_TIMEOUT = 600
```
Comment (oshtaier) [inline tests/foreman/ui/test_host.py]: probably need to return to previous value after sync is finished

File: tests/foreman/ui/test_host.py [hunk 5]
```diff
@@ -536,7 +545,284 @@ def test_positive_update_with_user(self):
...
+            query={
+                u'search': u'name={0}'.format(
+                    settings.server.hostname
+                )
+            }
+        )[0]
+        cls.proxy.location = [cls.loc]
+        cls.proxy = cls.proxy.update(['location'])
+        cls.proxy.organization = [cls.org]
+        cls.proxy = cls.proxy.update(['organization'])
```
Comment (oshtaier) [inline tests/foreman/ui/test_host.py]: `cls.proxy = cls.proxy.update(['organization', 'location'])`
and remove `cls.proxy = cls.proxy.update(['location'])` should work

**Comment by sghai** [comment]
Guys, Can we merge this PR, if you all are fine with code changes  ?

**Comment by rplevka** [comment]
ACK

**Comment by renzon** [comment]
ACK. I didn't merged because am not sure if @oshtaier comment regarding reverting values back is blocking this or not. So he check it up and merge.

**Comment by oshtaier** [comment]
There are still minor issues here, like comment that I left (can be improved in scope of some performance analysis) and created hostgroup that never used as far as I see, but that's fine
ACK

**Pull request merged**